### PR TITLE
Upgrade npm and bower dependencies to latest versions

### LINKED ISSUE #3887: Protractor tests don't work out-of-the-box with JHipster 3.5.0
From: https://groups.google.com/forum/?hl=en#!topic/jhipster-dev/0_KS33p0ETo

I ran the UI integration tests with `gulp itest`. I was surprised to have Firefox crash when Protractor started, so I updated Firefox to the latest version (47.0.1). This helped, but the tests still failed to run. I figured it'd be good to make sure Protractor was up-to-date, as well as other dependencies. I installed [npm-check-updates](https://github.com/tjunnone/npm-check-updates) using `npm install -g npm-check-updates`, then ran `ncu`.

```
$ ncu
⸨░░░░░░░░░░░░░░░░░░⸩ ⠇ :
 browser-sync                             2.12.10  →  2.13.0
 del                                        2.2.0  →   2.2.1
 eslint-plugin-angular                      1.0.1  →   1.3.0
 event-stream                               3.3.2  →   3.3.4
 gulp-angular-templatecache                 1.9.0  →   2.0.0
 gulp-eslint                                2.0.0  →   3.0.1
 gulp-flatten                               0.2.0  →   0.3.0
 gulp-imagemin                              3.0.1  →   3.0.2
 gulp-rev                                   7.0.0  →   7.1.0
 gulp-sass                                  2.3.1  →   2.3.2
 gulp-uglify                                1.5.3  →   1.5.4
 jasmine-reporters                          2.1.1  →   2.2.0
 karma                                    0.13.22  →   1.1.1
 karma-coverage                             1.0.0  →   1.1.0
 karma-junit-reporter                       1.0.0  →   1.1.0
 karma-phantomjs-launcher                   1.0.0  →   1.0.1
 protractor                                 3.3.0  →   4.0.0
 protractor-jasmine2-screenshot-reporter    0.3.1  →   0.3.2
 run-sequence                               1.2.1  →   1.2.2
 yargs                                      4.7.1  →   4.8.1

Run ncu with -u to upgrade package.json
```

Everything looked OK, so I ran `ncu -u`, followed by `npm install` and `gulp itest`. Firefox exhibited the same behavior as before (opened, but didn't connect to the application). To get Protractor working, I had to change a couple lines in `protractor.conf.js`.

```diff
-    seleniumServerJar: prefix + 'node_modules/protractor/selenium/selenium-server-standalone-2.52.0.jar',
+    seleniumServerJar: prefix + 'node_modules/protractor/node_modules/webdriver-manager/selenium/selenium-server-standalone-2.53.1.jar',

-    directConnect: true,
+    directConnect: false,
```

Finally, `gulp itest` passed with flying colors.

```
$ gulp itest
[08:18:35] Using gulpfile ~/dev/21-points/gulpfile.js
[08:18:35] Starting 'protractor'...
[08:18:35] W/configParser - pattern ./e2e/entities/*.js did not match any files.
[08:18:35] I/local - Starting selenium standalone server...
[08:18:35] I/launcher - Running 1 instances of WebDriver
[08:18:36] I/local - Selenium standalone server started at http://192.168.1.80:50537/wd/hub
Started
..........


10 specs, 0 failures
Finished in 29.031 seconds
[08:19:07] I/local - Shutting down selenium standalone server.
[08:19:07] I/launcher - 0 instance(s) of WebDriver still running
[08:19:07] I/launcher - firefox #01 passed
[08:19:07] Finished 'protractor' after 32 s
[08:19:07] Starting 'itest'...
[08:19:07] Finished 'itest' after 14 μs
```

### PR DESCRIPTION
**Pull request by mraible**
This also upgrades Protractor to 4.0 and fixes the out-of-the-box experience with `gulp itest`.

Fixes #3887.

### CODE REVIEW

**Comment by mraible** [comment]
Not sure why the one test is failing. @pascalgrimaud: any ideas?

**Comment by pascalgrimaud** [comment]
@mraible: sorry, I am on holidays, I cant have a look right now. I will try as soon as I m back

**Comment by deepu105** [comment]
could be a glitch. I have restarted that build lets see

File: generators/client/templates/_bower.json [hunk 1]
```diff
@@ -4,46 +4,46 @@
   "appPath": "<%= MAIN_SRC_DIR %>",
   "testPath": "<%= TEST_SRC_DIR %>spec",
   "dependencies": {
-    "angular": "1.5.5",
-    "angular-aria": "1.5.5",
-    "angular-bootstrap": "1.3.3",
+    "angular": "1.5.8",
+    "angular-aria": "1.5.8",
+    "angular-bootstrap": "2.0.0",
```
Comment (deepu105) [inline generators/client/templates/_bower.json]: this is a major version bump, we need to manually test the app to make sure stuff works as expected, coz our automated UI tests are not sufficient

File: generators/client/templates/_bower.json [hunk 2]
```diff
@@ -4,46 +4,46 @@
...
     <%_ } _%>
-    "swagger-ui": "2.1.4"
+    "swagger-ui": "2.1.5"
   },
   "devDependencies": {
-    "angular-mocks": "1.5.5"
+    "angular-mocks": "1.5.8"
   },
   "overrides": {
     "angular": {
```
Comment (deepu105) [inline generators/client/templates/_bower.json]: the jquery version here needs update as well

File: generators/client/templates/_bower.json (hunk 1 above)
Comment (mraible) [inline generators/client/templates/_bower.json]: @deepu105 I've confirmed this upgrade works by running the Protractor tests locally (using `gulp itest`) and having them pass. I also started the app and ensured there were no console errors.

File: generators/client/templates/_bower.json (hunk 2 above)
Comment (mraible) [inline generators/client/templates/_bower.json]: In `devDependencies` or `overrides`?

Comment (deepu105) [inline generators/client/templates/_bower.json]: Overrides. Coz it still has 2.x version there

File: generators/client/templates/_bower.json (hunk 1 above)
Comment (deepu105) [inline generators/client/templates/_bower.json]: Great

File: generators/client/templates/_bower.json (hunk 2 above)
Comment (deepu105) [inline generators/client/templates/_bower.json]: I added this

**Comment by deepu105** [comment]
I cancelled the PR build since there is a commit build on it already coz the branch here is on the main repo. @mraible you need to create the fix branches on your fork and PR to jhipster repo to avoid double travis runs.

Anyways there seems to some issue with the no i18n build coz protractor tests are always failing there and I have no idea why

**Comment by mraible** [comment]
@deepu105 Are you saying you want me to re-create this PR by forking and creating a branch on my fork?

**Comment by deepu105** [comment]
Not this one, im just saying generally to save some travis time :)

Thanks & regards,
Deepu

**Comment by mraible** [comment]
Closed and re-opened to (hopefully) trigger a CI build.

**Pull request merged**